hydroflow_plus/src/location/tick.rs

```diff
@@ -53,13 +94,20 @@ impl<'a, L: Location<'a>> Tick<L> {
         batch_size: impl QuotedWithContext<'a, usize, L> + Copy + 'a,
     ) -> Stream<(), Self, Bounded>
     where
-        L: NoTick,
+        L: NoTick + NoTimestamp,
     {
-        self.l
+        let out = self
+            .l
             .spin()
-            .flat_map(q!(move |_| 0..batch_size))
+            .flat_map_ordered(q!(move |_| 0..batch_size))
             .map(q!(|_| ()))
-            .tick_batch(self)
+            .timestamped(self);
+
+        unsafe {
+            // SAFETY: at runtime, `spin` produces a single value per tick,
+            // so each batch is guaranteed to be the same size.
+            out.tick_batch()
+        }
```

Comment:
For now can just leave the rest of the unsafe blocks as is, but would be good as a convention to only wrap the unsafe method calls in `unsafe`, otherwise the whole call chain goes into `unsafe` and it doesn't distinguish anything (have to check each method to see if that is the unsafe one)

Reply:
Yeah, I think I will try to write up a style guide soon on what to wrap in `unsafe`. Sometimes it's nice to include additional API calls if it is a "temporary"/"local" unsafety rather than one which propagates out the function call.